src/v/cloud_storage_clients/xml_sax_parser.cc

```diff
@@ -116,7 +116,7 @@ void aws_parse_impl::handle_characters(std::string_view characters) {
     switch (_current_tag) {
     case xml_tag::key:
         if (_current_item) {
-            _current_item->key = {characters.data(), characters.size()};
+            _current_item->key += {characters.data(), characters.size()};
```

Comment:
`+=` takes only a `ss::sstring`: https://github.com/scylladb/seastar/blob/99e07a1d921f46cf5d68e1f46776fa81a26a0158/include/seastar/core/sstring.hh#L681

This will cause two copies. Can we just use `append` that will prevent an extra copy.


Also `std::string` would be better in this case. `ss::sstring` does not have a a capacity so every `append` will be 1 allocation. Maybe that's fine, but `std::string` is like `std::vector` in that it can allocate a greater size... IDK I'm probably over optimizing at this point. Let's keep the `ss::string` and switch to `append`.